static/js/message_util.js

```diff
@@ -16,6 +16,23 @@ function add_messages(messages, msg_list, opts) {
     return render_info;
 }
 
+function does_message_content_have_html_tag(message, tagName) {
+    // creating a virtual dom to check for html tag in message.content.
+    return $('<div>').append(message.content).find(`${tagName}:first`).length > 0;
```

Comment:
I'm curious about GC related issues with this. I added a similar check in one of my PRs but I've a feeling JQuery might keep a reference to the created object internally and as a result we could end up with a lot of Query objects each time this function is called.

Do you know if that isn't the case here? I couldn't find anything relevant online so far.

Reply:
I will write a quick jsfiddle and monitor it's memory usage to see if it's the case.

Reply:
https://github.com/jquery/jquery/blob/master/src/core/init.js
https://github.com/jquery/jquery/blob/master/src/core.js

I think it just return object and it should be garbage collected.